### PR TITLE
Fix incorrect delete in MXExecutorReshape exception handling

### PR DESCRIPTION
**Pull request by taliesinb**
MXExecutorReshape makes a very obvious and simple error in its cleanup code, which is that it deletes the location of the out pointer rather the pointed-to executor. It also does this deletion even if the out pointer was not set, so it could delete a nullptr or other garbage value.

### CODE REVIEW

**Comment by marcoabreu** [comment]
@mkolod @KellenSunderland

**Comment by KellenSunderland** [comment]
Taking a look ...

**Comment by KellenSunderland** [comment]
So I see we potentially are deleting memory we shouldn't whenever an error is encountered.  What I'm wondering is what would happen if an error is now encountered on line 569 for example?

**Comment by stu1130** [comment]
@mxnet-label-bot add [pr-awaiting-review]
Thanks for your contribution @taliesinb

**Comment by taliesinb** [comment]
@KellenSunderland good point. I've moved the nullptr set up to just after API_BEGIN.

File: src/c_api/c_api_executor.cc [hunk 1]
```diff
@@ -560,6 +560,7 @@ int MXExecutorReshape(int partial_shaping,
                       ExecutorHandle *out) {
   MXAPIThreadLocalEntry *ret = MXAPIThreadLocalStore::Get();
   API_BEGIN();
+  *out = nullptr; // ensure we can know whether to free executor on early abort
```
Comment (KellenSunderland) [inline src/c_api/c_api_executor.cc]: Small linting issue:
src/c_api/c_api_executor.cc:563:  At least two spaces is best between code and comments  [whitespace/comments] [2]
Category 'whitespace' errors found: 1

Comment (taliesinb) [reply src/c_api/c_api_executor.cc]: I've added the space, and am also creating a temporary pointer for the new executor as I see is done in other functions like MXExecutorGetOptimizedSymbol.

**Comment by taliesinb** [comment]
@KellenSunderland Good to merge?

**Comment by roywei** [comment]
@anirudh2290 @apeforest could you help take a look and merge if looks good? Thanks!

**Comment by roywei** [comment]
@mxnet-label-bot add[C API, pr-awaiting-merge]

**Comment by KellenSunderland** [comment]
@taliesinb Sorry for the delay, looks good.  Thanks for the fix!

**Pull request merged**